### PR TITLE
Repoman - Adding relative base path ./ to path rewrites.

### PR DESCRIPTION
**Pull request by hattan**
This PR addresses part of #524 , specifically adding ./ to local paths. This is needed to support Terraform local modules PR #532. A separate PR will be provided to add the ReWrite Rules section.

This PR introduces:

- A utility method in repoman utils to add a base path that is platform specific "./" or ".\\".
- Changes to the repoman path rewrite normalization to add a base prefix.

Tests were conducted to verify the CI workflow for repoman generate and a test azd deployment was conducted against a test repo with the relative base path included "./"

[Repo generation CI (on fork)](https://dev.azure.com/csedevops/Azure-dev/_build/results?buildId=15093&view=logs&s=2f4b62b5-8d35-5570-0513-04a0c5199809
)

Resources Deployed via modified template:
image

Running Application:
image

### CODE REVIEW

File: generators/repo/src/common/util.ts [hunk 1]
```diff
@@ -66,6 +66,17 @@ export const cleanDirectoryPath = async (directoryPath: string, cleanGit: boolea
     await del(patterns, { cwd: directoryPath, force: true, dot: true });
 }
 
+export const ensureRelativeBasePath  = (input : string) => {
+    const basePath =`.${path.sep}`
+    if(!input.startsWith(basePath) && !input.startsWith(`.${basePath}`)) {
+        input = `${basePath}${input}`
+    }
+    else{
+        console.warn(chalk.yellowBright(` - ${input} already contains a relative base path.`));
```
Comment (wbreza) [inline generators/repo/src/common/util.ts:75]: nit: May want to only write out if `--debug` flag is set?

**Review by wbreza — APPROVED** [review]
LGTM

File: generators/repo/src/common/util.ts (hunk 1 above)
Comment (hattan) [reply generators/repo/src/common/util.ts:75]: Thanks for the feedback, I kept the logic similar to the original design which was to keep a basic output message with the change to a file, but if --debug is set it displays a lot more details about individual substitutions in a file.

**Review by ellismg — APPROVED** [review]
LGTM

**Pull request merged**